ios/Codeforces Watcher/Application/features/auth/verify/VerifyViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  VerifyViewController.swift
+//  Codeforces Watcher
+//
+//  Created by Ivan Karavaiev on 2/15/21.
+//  Copyright © 2021 xorum.io. All rights reserved.
+//
+
+import UIKit
+import common
+import PKHUD
+
+class VerifyViewController: ClosableViewController, ReKampStoreSubscriber {
+    
+    private let contentView = UIView()
+    
+    private let handleInput = TextInputLayout(hint: "codeforces_handle".localized, type: .email)
+    private let instructionLabel = VerifyInstructionLabel(text: "verify_instruction".localized)
+    private let verificationCodeLabel = UILabel().apply {
+        $0.textAlignment = .center
+    }
+    private let hintLabel = HintLabel().apply {
+        $0.text = "verify_change_it_back".localized;
+    }
+    private let verifyButton = PrimaryButton().apply {
+        $0.setTitle("verify".localized.uppercased(), for: .normal)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        store.subscribe(subscriber: self) { subscription in
+            subscription.skipRepeats { oldState, newState in
+                return KotlinBoolean(bool: oldState.verification == newState.verification)
+            }.select { state in
+                return state.verification
+            }
+        }
+    }
+    
+    func onNewState(state: Any) {
+        let state = state as! VerificationState
+        
+        switch (state.status) {
+        case .done:
```

Comment:
IMHO, it's better to use order, which naturally occurs `idle -> pending -> done`